contracts/Datastore.sol

```diff
@@ -420,19 +481,33 @@ contract Datastore is AragonApp {
      * @param _groupWrite Write permission
      * @param _entities Ids of the groups
      * @param _entityRead Read permission
-     * @param _entityWrite Write permission      
+     * @param _entityWrite Write permission
+     * @param _isPublic Public status
+     * @param _storageRef Storage reference
+     * @param _fileSize File size
+     * @param _encryptionKey Encryption key
      */
-    function setMultiplePermissions(uint256 _fileId, uint256[] _groupIds, bool[] _groupRead, bool[] _groupWrite, address[] _entities, bool[] _entityRead, bool[] _entityWrite, bool _isPublic) public onlyFileOwner(_fileId) {
-        
+    function setMultiplePermissions(
+        uint256 _fileId, uint256[] _groupIds, bool[] _groupRead, bool[] _groupWrite, 
+        address[] _entities, bool[] _entityRead, bool[] _entityWrite, bool _isPublic, string _storageRef, 
+        uint128 _fileSize, string _encryptionKey) 
+        public 
+        onlyFileOwner(_fileId) 
+    {
         for(uint256 i = 0; i < _groupIds.length; i++) 
             permissions.setGroupPermissions(_fileId, _groupIds[i], _groupRead[i], _groupWrite[i]);
         
         for(uint256 j = 0; j < _entities.length; j++) 
             permissions.setEntityPermissions(_fileId, _entities[j], _entityRead[j], _entityWrite[j]);
 
-        fileList.files[_fileId].isPublic = _isPublic;
+        fileList.setPublic(_fileId, _isPublic);
+
+        if (!_isPublic || (_isPublic && keccak256(_encryptionKey) == keccak256(""))) {
```

Comment:
Not sure if I understand this correctly. What does exactly `_isPublic` mean?

Reply:
Public files will be unencrypted, and therefore readable by anyone inside or outside the DAO. Private files will be encrypted and only the addresses/groups specified will be able to access it's content.

Reply:
Ok, I don't understand the purpose of the condition yet.
- First part would mean that file is private. Then you are setting the encryption key. But if it was private I guess it already had a key. What happens then? Is the file going to be unencrypted with the old key and re-encrypted with the new one? Otherwise it could become unreadable, right?
- Second part holds if it's public and it has an empty encryption key, which sounds redundant: why should it have a key if it's unencrypted? Anyway, then you are setting the key. What for if it's public, i.e., unencrypted?

Reply:
Yes there seems to be something wrong with this function. We should have more details monday after our team meeting.

Reply:
@macor161 cherry-pick this one here too, please

Reply:
Done!